testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpServerTest.java

```diff
@@ -0,0 +1,709 @@
+            spanAssertions.add(
+                span -> assertIndexedServerSpan(span, requestId).hasParent(rootSpan));
+
+            if (options.hasHandlerSpan.apply(endpoint)) {
+              spanAssertions.add(
+                  span -> assertHandlerSpan(span, "GET", endpoint).hasParent(trace.getSpan(1)));
+            }
+
+            int parentIndex = spanAssertions.size() - 1;
+            spanAssertions.add(
+                span ->
+                    assertIndexedControllerSpan(span, requestId)
+                        .hasParent(trace.getSpan(parentIndex)));
+
+            trace.hasSpansSatisfyingExactly(spanAssertions);
+          });
+    }
+    testing.waitAndAssertTraces(assertions);
+  }
+
+  // NOTE: this method does not currently implement asserting all the span types that groovy
+  // HttpServerTest does
+  protected void assertTheTraces(
+      int size,
+      String traceId,
+      String parentId,
+      String method,
+      ServerEndpoint endpoint,
+      String errorMessage,
+      AggregatedHttpResponse response) {
+    List<Consumer<TraceAssert>> assertions = new ArrayList<>();
+    for (int i = 0; i < size; i++) {
+      assertions.add(
+          trace -> {
+            List<Consumer<SpanDataAssert>> spanAssertions = new ArrayList<>();
+            spanAssertions.add(
+                span -> {
+                  assertServerSpan(span, method, endpoint);
+                  if (parentId != null) {
+                    span.hasTraceId(traceId).hasParentSpanId(parentId);
+                  } else {
+                    span.hasNoParent();
+                  }
+                });
+
+            if (options.hasHandlerSpan.apply(endpoint)) {
+              spanAssertions.add(
+                  span -> {
+                    assertHandlerSpan(span, method, endpoint);
+                    span.hasParent(trace.getSpan(0));
+                  });
+            }
+
+            if (endpoint != NOT_FOUND) {
+              int parentIndex = spanAssertions.size() - 1;
+              spanAssertions.add(
+                  span -> {
+                    assertControllerSpan(span, errorMessage, options.expectedExceptionClass);
+                    span.hasParent(trace.getSpan(parentIndex));
+                  });
+            }
+
+            if (options.hasResponseSpan.apply(endpoint)) {
+              int parentIndex = spanAssertions.size() - 1;
+              spanAssertions.add(
+                  span -> {
+                    assertResponseSpan(span, method, endpoint);
+                    span.hasParent(trace.getSpan(parentIndex));
+                  });
+            }
+
+            if (options.hasErrorPageSpans.apply(endpoint)) {
+              spanAssertions.addAll(errorPageSpanAssertions(method, endpoint));
+            }
+
+            trace.hasSpansSatisfyingExactly(spanAssertions);
+          });
+    }
+
+    testing.waitAndAssertTraces(assertions);
+  }
+
+  public SpanDataAssert assertControllerSpan(
```

Comment:
Let's make `protected` where possible. Also surprised errorprone isn't suggesting making these static, thought we had that configured